controllers/drplacementcontrol_controller.go

```diff
@@ -1179,12 +1179,8 @@ func (r *DRPlacementControlReconciler) processDeletion(ctx context.Context,
 	}
 
 	if placementObj != nil && controllerutil.ContainsFinalizer(placementObj, DRPCFinalizer) {
-		// Remove DRPCFinalizer from User PlacementRule/Placement.
-		controllerutil.RemoveFinalizer(placementObj, DRPCFinalizer)
-
-		err := r.Update(ctx, placementObj)
-		if err != nil {
-			return fmt.Errorf("failed to update User PlacementRule/Placement %w", err)
+		if err := r.finalizePlacement(ctx, placementObj); err != nil {
+			return err
```

Comment:
@nirs This change contradicts with DRPC finalization logic. Function finalizeDRPC is not removing the finalizer, it is running some logic. And then finalizer is deleted outside this function. Here you are removing placement finalizer inside finalizePlacement function

Reply:
This is fine, finalizing drpc is a special case when we must do all the changes and then finalize the placement, and only if the finalizing the placement succeeded we can go on and remove the drpc finalizer. The change keeps the exact behavior, just moving the details outside to keep the code cleaner and easier to modify in the future.

Reply:
@ELENAGER we can change finalizeDRPC to prepareForFinalizeDRPC and add finalizeDRPC that will remove the finalizer and update the resource. This way finalizeDRPC and finalizePlacement behave in the same way. What do yo think?